src/Extractor/ResponseRpcExtractor.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\JsonRpc\Extractor;
+
+use Psr\Http\Message\ResponseInterface;
+
+final class ResponseRpcExtractor extends RpcExtractor
+{
+    public static function getRpcErrorCode(ResponseInterface $response) : ?int
+    {
+        $error = self::getFieldFromBody($response, 'error');
+
+        return $error['code'] ?? null;
+    }
+
+    public static function getRpcErrorMessage(ResponseInterface $response) : ?string
+    {
+        $error = self::getFieldFromBody($response, 'error');
+
+        return $error['message'] ?? null;
+    }
+
+    /**
+     * @return array<array<array<string>>>|null
+     */
+    public static function getRpcErrorData(ResponseInterface $response) : ?array
```

Comment:
nemusí to být array, může to být cokoliv

Reply:
ja jsem koukal co to vraci, davalo to jen array

Reply:
to je protože jsi to dělal podle jednoho konkrétního API, který se tak chová ... to neznamená že všechny json-rpc API to tak dělají